Source/core/SocketPort.cpp

```diff
@@ -1144,11 +1144,11 @@ namespace Thunder {
         }
 
         /* virtual */ int32_t SocketPort::Read(uint8_t buffer[], const uint16_t length) const {
-            return (::recv(m_Socket, reinterpret_cast<char*>(buffer), length, 0));
+            return (::recv(m_Socket, reinterpret_cast<char*>(buffer), length, MSG_NOSIGNAL));
```

Comment:
This should be thoroughly tested on production systems or made an option to enable it on a created SocketPort as this might cause regression. 
I wonder if this is related to the investgation @sebaszm did with the closedown of the socket. @sebaszm to comment?
I guess this might change the flow of action in closing down "failing/closing" sockets...

Reply:
I don't think MSG_NOSIGNAL is valid on recv() (so this has a chance to fail with EINVAL).

Reply:
@pwielders, @sebaszm, good observation. I have checked and the flag has not been defined in that context.